File: app/src/main/java/com/esorokin/lantector/model/network/api/handler/InternalCompletableResponseHandler.java
package com.esorokin.lantector.model.network.api.handler;

import retrofit2.Response;

class InternalCompletableResponseHandler implements HttpResponseAdapter<Object> {
	@Override
	public Object adaptHttpResponse(Response<Object> response) {
		if (response.isSuccessful()) {
			return response;
		} else {
			throw new RuntimeException(response.message() + "");
		}
	}
}

File: app/src/main/java/com/esorokin/lantector/model/network/api/handler/InternalCompletableResponseHandler.kt
package com.esorokin.lantector.model.network.api.handler

import retrofit2.Response

internal class InternalCompletableResponseHandler : HttpResponseAdapter<Any> {
    override fun adaptHttpResponse(response: Response<Any>): Any {
        if (response.isSuccessful) {
            return response
        } else {
            throw RuntimeException(response.message() + "")
        }
    }
}
